client/src/main.rs

```diff
@@ -12,205 +12,271 @@
+/// Execute the actions, commands, and queries in the file at `query_file_path`.
+fn file(
+    runtime: Runtime,
+    mut flight_service_client: FlightServiceClient<Channel>,
+    query_file_path: &str,
+) -> Result<(), Box<dyn Error>> {
+    let file = File::open(query_file_path)?;
     let lines = io::BufReader::new(file).lines();
 
     for line in lines {
-        // Removes all comments.
+        // Remove any comments.
         let input = line?;
         let query = if let Some(comment_start) = input.find("--") {
             input[0..comment_start].to_owned()
         } else {
             input
         };
 
-        // Executes the query.
+        // Execute the query.
         if !query.is_empty() {
             println!("{}", query);
-            if let Err(message) = execute_and_print_action_command_or_query(&rt, &mut fsc, &query) {
+            if let Err(message) = execute_and_print_action_command_or_query(
+                &runtime,
+                &mut flight_service_client,
+                &query,
+            ) {
                 eprintln!("{}", message);
             }
-            println!(); // Formatting newline.
+            // Formatting newline.
+            println!();
         }
     }
+
     Ok(())
 }
 
-fn repl(rt: Runtime, mut fsc: FlightServiceClient<Channel>) -> Result<()> {
+/// Execute actions, commands and queries in a read-eval-print loop.
+fn repl(
+    runtime: Runtime,
+    mut flight_service_client: FlightServiceClient<Channel>,
+) -> Result<(), Box<dyn Error>> {
+    // Create the read-eval-print loop.
     let mut editor = Editor::<ClientHelper>::new()?;
-    let table_names = retrieve_table_names(&rt, &mut fsc)?;
+    let table_names = retrieve_table_names(&runtime, &mut flight_service_client)?;
     editor.set_helper(Some(ClientHelper::new(table_names)));
+
+    // Read previously executed actions, commands, and queries from the history file.
     let history_file_name = ".modelardb_history";
     if let Some(mut home) = dirs::home_dir() {
         home.push(history_file_name);
         let _ = editor.load_history(&home);
     }
 
+    // Execute actions, commands, and queries and print the result.
     while let Ok(line) = editor.readline("ModelarDB> ") {
         editor.add_history_entry(line.as_str());
-        if let Err(message) = execute_and_print_action_command_or_query(&rt, &mut fsc, &line) {
+        if let Err(message) =
+            execute_and_print_action_command_or_query(&runtime, &mut flight_service_client, &line)
+        {
             eprintln!("{}", message);
         }
     }
 
+    // Append the executed actions, commands, and queries to the history file.
     if let Some(mut home) = dirs::home_dir() {
         home.push(history_file_name);
         let _ = editor.append_history(&home);
     }
+
     Ok(())
 }
 
+/// Execute an action, a command, or a query. Returns [`Error`] if the action, command, or query
+/// could not be executed and their result retrieved.
```

Comment:
Should this be `...or their result could not be retrieved.`?

Reply:
Fixed.